File: src/ui/gfx/frame/Frame.java
package ui.gfx.frame;

import lombok.Getter;
import lombok.RequiredArgsConstructor;

@RequiredArgsConstructor
public class Frame {
    @Getter
    private final String name;

    public final int x, y;
    public final int width, height;
    public final int offsetX, offsetY;

    public Frame(String name, int x, int y, int width, int height, int offsetX) {
        this(name, x, y, width, height, offsetX, 0);
    }

    public Frame(String name, int x, int y, int width, int height) {
        this(name, x, y, width, height, 0, 0);
    }
}

File: src/main/kotlin/ui/gfx/frame/Frame.kt
package ui.gfx.frame

class Frame(val name: String, val x: Int, val y: Int, val width: Int, val height: Int, val offsetX: Int, val offsetY: Int) {

    constructor(name: String, x: Int, y: Int, width: Int, height: Int) :
            this(name, x, y, width, height, 0, 0)

    constructor(name: String, x: Int, y: Int, width: Int, height: Int, offsetX: Int) :
            this(name, x, y, width, height, offsetX, 0)
}
